plugins/hls-stan-plugin/src/Ide/Plugin/Stan.hs

```diff
@@ -80,7 +81,7 @@ rules recorder plId = do
   define (cmapWithPrio LogShake recorder) $
     \GetStanDiagnostics file -> do
       config <- getPluginConfigAction plId
-      if pluginEnabledConfig plcDiagnosticsOn config then do
+      if pluginEnabledConfig plcDiagnosticsOn config  == PluginEnabled then do
```

Comment:
This sort of thing would look nicer if we split the "plugin is globally enabled" method from the "plugin wants to respond to this request" method. Then it would be sensible to generally call the "globally enabled" method from rules, since it of course doesn't look at the current request. And then we wouldn't be at risk of this getting out of sync.